src/libaktualizr/utilities/utils.cc

```diff
@@ -327,13 +328,19 @@ void Utils::writeFile(const boost::filesystem::path &filename, const std::string
     throw std::runtime_error(std::string("Error opening file ") + tmpFilename.string());
   }
   file << content;
+  if (atomic) {
```

Comment:
Why do we need sync calls to make it atomic?

Reply:
Ok the first sync operation is probably needed to be safe. However:

- that function was written to be atomic in the first place, so I don't think we need the boolean flag
- `sync()` is expensive if we call it every time we create a file. Calling `fsync()` would be ideal but looks difficult/impossible on a standard ofstream. Maybe we'd have to resort to boost (https://stackoverflow.com/questions/676787/how-to-do-fsync-on-an-ofstream) or POSIX C
- renaming a file on the same filesystem should be atomic, so I don't see the point of the `sync()` after rename